pandas/core/nanops.py

```diff
@@ -463,6 +463,7 @@ def newfunc(values: np.ndarray, *, axis: int | None = None, **kwargs):
             axis == 1
             and values.ndim == 2
             and values.flags["C_CONTIGUOUS"]
+            and (values.shape[1] / values.shape[0]) > 1000
```

Comment:
can you add some comments about why these things are needed (e.g. for performance), can ref this issue as well.

Reply:
Added a comment with reference to my timings and update the docstring